raiden/transfer/channel.py

```diff
@@ -1310,11 +1312,15 @@ def events_for_close(
             None,
         )
 
+        balance_proof = channel_state.partner_state.balance_proof
+        # silence mypy: partner's balance proofs should be signed
```

Comment:
IMO the correct fix is to split the `NettingChannelEndState` class into two, one for our state and another for the partner's state. If we do split it, once we have annotations for most of the code base we can then remove the runtime type assertions. (Note: As far as I recall we only have not done it yet to safe time)

Reply:
Created an issue for that https://github.com/raiden-network/raiden/issues/3222